### PR TITLE
Add Action to run Policy Framework tests on PRs

### PR DESCRIPTION
**Pull request by JustinKuli**
This will just run the e2e/kind tests from the policy framework repo -
performing some basic validation that all the policy components will
still work well together with any changes here.

Refs:
 - stolostron/backlog#21795

Signed-off-by: Justin Kulikauskas <jkulikau@redhat.com>

### CODE REVIEW

**Review by mprahl — APPROVED** [review]
/hold for other reviews if desired

File: .github/workflows/framework-kind.yml
Comment (JustinKuli) [reply .github/workflows/framework-kind.yml:7]: I'll cherry-pick these to 2.5, I want to add similar things to 2.4 as well, but they might require some other tweaks so I chose 2.5+ for now.

**Comment by JustinKuli** [comment]
Great suggestion on the `exclude`! If the matrix gets too huge (maybe with more kind versions or something), that would be good to keep things reasonable. But, GitHub Actions are free for public repos, so I don't think we need to worry about costs associated with the size of the matrix.

**Comment by dhaiducek** [comment]
I was also wondering if these should be `Required` as status checks, but if these ever fail due to equivalent changes required in a different repo, it might be good to keep these as informational?

**Comment by mprahl** [comment]
> I was also wondering if these should be `Required` as status checks, but if these ever fail due to equivalent changes required in a different repo, it might be good to keep these as informational?

The intention is that it if the framework tests fail, the automatic cherry pick PRs from the upcoming cherry pick bot won't be automatically merged. We could have the bot verify that the "informational" checks passed before merging but I'd like to make this required for now.

**Comment by dhaiducek** [comment]
I mention it partially because it's managed in Prow:
https://github.com/openshift/release/blob/master/core-services/prow/02_config/stolostron/governance-policy-status-sync/_prowconfig.yaml

**Comment by JustinKuli** [comment]
> I mention it partially because it's managed in Prow: https://github.com/openshift/release/blob/master/core-services/prow/02_config/stolostron/governance-policy-status-sync/_prowconfig.yaml

Oh yeah, that's where that is! I know I've found that before! Speaking of which, looks like we should add that for release-2.6. In all of those files. 😅 

Personally I'd rather manage it through GitHub, instead of another config file in another repo (in another folder, as opposed to [ci-opertator/config/stolostron](https://github.com/openshift/release/tree/master/ci-operator/config/stolostron) where I usually look)

**Comment by dhaiducek** [comment]
>Personally I'd rather manage it through GitHub, instead of another config file in another repo (in another folder, as opposed to [ci-opertator/config/stolostron](https://github.com/openshift/release/tree/master/ci-operator/config/stolostron) where I usually look)

We can try setting it here, but I think Prow will override it (since it needs to make sure its jobs are marked as required also).

>Speaking of which, looks like we should add that for release-2.6. In all of those files. 😅

Shoot--you're right! I thought the script would have taken care of that. I'll add it to my to-dos.

**Pull request merged**
